apps/emqx_ai_completion/src/emqx_ai_completion_schema.erl

```diff
@@ -0,0 +1,219 @@
+            mk(binary(), #{
+                required => true,
+                desc => ?DESC(provider_name),
+                validator => fun validate_name/1
+            })},
+        {type,
+            mk(hoconsc:enum([openai, anthropic]), #{
+                default => openai, required => true, desc => ?DESC(type)
+            })},
+        {api_key, emqx_schema_secret:mk(#{required => true, desc => ?DESC(api_key)})},
+        {transport_options,
+            mk(ref(transport_options), #{
+                default => #{},
+                desc => ?DESC(transport_options),
+                importance => ?IMPORTANCE_HIDDEN
+            })}
+    ];
+fields(transport_options) ->
+    [
+        {connect_timeout,
+            mk(emqx_schema:timeout_duration_ms(), #{
+                required => false,
+                default => <<"1s">>,
+                desc => ?DESC(connect_timeout),
+                importance => ?IMPORTANCE_HIDDEN
+            })},
+        {recv_timeout,
+            mk(emqx_schema:timeout_duration_ms(), #{
+                required => false,
+                default => <<"5s">>,
+                desc => ?DESC(recv_timeout),
+                importance => ?IMPORTANCE_HIDDEN
+            })},
+        {checkout_timeout,
+            mk(emqx_schema:timeout_duration_ms(), #{
+                required => false,
+                default => <<"1s">>,
+                desc => ?DESC(checkout_timeout),
+                importance => ?IMPORTANCE_HIDDEN
+            })}
+    ];
+fields(provider_api_get) ->
+    without_fields([api_key], fields(provider));
+fields(provider_api_put) ->
+    without_fields([name], fields(provider));
+fields(openai_completion_profile) ->
+    [
+        {name,
+            mk(binary(), #{
+                required => true,
+                desc => ?DESC(completion_profile_name),
+                validator => fun validate_name/1
+            })},
+        {type, mk(openai, #{default => openai, required => true, desc => ?DESC(type)})},
+        {provider_name, mk(binary(), #{required => true, desc => ?DESC(provider_name)})},
+        {system_prompt, mk(binary(), #{default => <<>>, desc => ?DESC(system_prompt)})},
+        {model, mk(binary(), #{default => <<"gpt-4o">>, desc => ?DESC(model)})}
+    ];
+fields(openai_completion_profile_api_get) ->
+    fields(openai_completion_profile);
+fields(openai_completion_profile_api_put) ->
+    without_fields([name], fields(openai_completion_profile));
+fields(anthropic_completion_profile) ->
+    [
+        {name,
+            mk(binary(), #{
+                required => true,
+                desc => ?DESC(completion_profile_name),
+                validator => fun validate_name/1
+            })},
+        {type, mk(anthropic, #{default => anthropic, required => true, desc => ?DESC(type)})},
+        {provider_name, mk(binary(), #{required => true, desc => ?DESC(provider_name)})},
+        {anthropic_version,
+            mk(enum(['2023-06-01']), #{
```

Comment:
Q: should we be strict about this version, or let it fail in runtime (similar to the comment for `model`)?

Reply:
I think yes, because this value defines the shape of the used API (methods, data structs), so we cannot freely switch to other versions without changing the code.